src/lib/optimizer/strategy/predicate_placement_rule.hpp

```diff
@@ -12,7 +12,8 @@ class PredicateNode;
 
 /**
  * Heuristic rule pushing non-expensive predicates down as far as possible (to reduce the result set early on) and
- * pulling expensive predicates up as far as possible.
+ * pulling expensive predicates up as far as possible. For the purpose of this rule, semi- and anti-joins are treated
```

Comment:
Yes, but for most queries, that should not apply any more:

> We (heuristically) consider a predicate to be expensive if it contains a correlated subquery. Otherwise, we consider it to be cheap.